Esterni/ManualeSviluppatore/components/c5-Architettura.tex

```diff
@@ -0,0 +1,74 @@
+\section{Architettura}
+\subsection{Pattern architetturale: MVC}
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale = 1]{components/img/mvc-generico.png}
+    \caption{Diagramma pattern MVC}
+    \label{fig:diagramma MVC generico}
+\end{figure}
+L'architettura del capitolato C7 aderisce al pattern architetturale \textit{Model-View-Controller (MVC).} L'aderenza al pattern era un requisito obbligatorio imposto dall'azienda ma sono stati riscontrati diversi punti positivi nella sua implementazione:
+
+\begin{itemize}
+    \item Rapidità nello sviluppo software dato dalla possibilità di lavoro parallelo su modello, vista e controller;
+    \item Indipendenza dei vari componenti;
+    \item Maggior facilità nella testabilità, manutenibilità e scalabilità.
+\end{itemize}
+
+\subsection{Diagramma dei pacchetti}
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale = 0.5]{components/img/diagramma-package.png}
+    \caption{Diagramma pacchetti}
+    \label{fig:diagramma pacchetti}
+\end{figure}
+\subsection{Diagramma delle classi - model}
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale = 0.6]{components/img/diagramma-classi-model.png}
+    \caption{Diagramma delle classi nel model}
+    \label{fig:Diagramma delle classi nel model}
+\end{figure}
+
+\subsection{Design pattern creazionali: Singleton}
+
+Le classi mostrate in figura 3 rispettano il pattern singleton, il che le porta ad avere una sola loro istanza all'interno del programma.
```

Comment:
in figura 3 metterei il riferimento:

in figura \ref{fig:Diagramma delle classi nel model}